### PR TITLE
Adding ability to add extra volumes, volumeMounts and initContainers

### PR DESCRIPTION
**Pull request by swade1987**
This PR enables users to add extra volumes, volume mounts and initContainer's to the chart deployed to their cluster if required.

### CODE REVIEW

**Comment by stefanprodan** [comment]
@swade1987 the chart is invalid as per e2e tests: `Error: release flux failed: Deployment.apps "flux" is invalid: spec.template.spec.containers: Required value`

File: chart/flux/values.yaml [hunk 1]
```diff
@@ -214,3 +219,13 @@ extraEnvs: []
 
 prometheus:
   enabled: false
+
+# Add your own init container or uncomment and modify the given example.
+initContainers: {}
+  # flux-init:  # <- will be used as container name
+  #   image: "busybox:1.30.1"
+  #   imagePullPolicy: "IfNotPresent"
+  #   command: ['sh', '-c', 'counter=0; until [ "$counter" -ge 30 ]; do if [ -f /tmp/flux-deploy-key/xyz.key ]; then exit 0; else echo waiting for flux deploy key && sleep 1 && counter=$((counter+1)); fi; done; exit 1;']
```
Comment (stefanprodan) [inline chart/flux/values.yaml]: Can we make this a valid example for the default sync key?

File: chart/flux/templates/deployment.yaml [hunk 2]
```diff
@@ -183,3 +195,4 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
+  {{- end }}
```
Comment (stefanprodan) [inline chart/flux/templates/deployment.yaml]: Add a new line at the end please.

File: chart/flux/values.yaml [hunk 3]
```diff
@@ -214,3 +219,13 @@ extraEnvs: []
...
+  #   volumeMounts:
+  #     - mountPath: /tmp/flux-deploy-key
+  #       name: flux-deploy-key
```
Comment (stefanprodan) [inline chart/flux/values.yaml]: Add a new line at the end please.

**Comment by squaremo** [comment]
First-time contributor high five :hand:

**Comment by swade1987** [comment]
@squaremo thanks man, just trying to get these e2e tests working 😢

File: chart/flux/templates/deployment.yaml [hunk 4]
```diff
@@ -69,6 +69,15 @@ spec:
...
       {{- end }}
+      {{- if .Values.extraVolumes }}
+{{ toYaml .Values.extraVolumes | indent 8 }}
+      {{- end }}
+{{- if .Values.initContainers }}
+      initContainers:
+{{- range $key, $value := .Values.initContainers }}
+        - name: {{ $key }}
+{{ toYaml $value | indent 10 }}
+{{- end }}
```
Comment (squaremo) [inline chart/flux/templates/deployment.yaml]: Is there a missing `{{- end }}` here (for the `if .Values.initContainers`)?

Comment (swade1987) [reply chart/flux/templates/deployment.yaml]: Added this but still not happy.

File: chart/flux/templates/deployment.yaml [hunk 5]
```diff
@@ -183,3 +196,4 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
+  {{- end }}
```
Comment (squaremo) [inline chart/flux/templates/deployment.yaml]: This `{{ - end }}` doesn't appear to match anything

File: .idea/flux.iml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (squaremo) [inline .idea/flux.iml]: (accidental addition of IDE files)

Comment (swade1987) [reply .idea/flux.iml]: Fixed, my bad!

File: chart/flux/templates/deployment.yaml (hunk 5 above)
Comment (swade1987) [reply chart/flux/templates/deployment.yaml]: Done.

File: chart/flux/values.yaml (hunk 3 above)
Comment (swade1987) [reply chart/flux/values.yaml]: Done.

File: chart/flux/templates/deployment.yaml (hunk 2 above)
Comment (swade1987) [reply chart/flux/templates/deployment.yaml]: Done.

File: chart/flux/values.yaml (hunk 1 above)
Comment (swade1987) [reply chart/flux/values.yaml]: Done.

**Comment by 2opremio** [comment]
It seems all comments have been addressed and I don't have any myself. LGTM

**Comment by 2opremio** [comment]
@swade1987 can you please squash the changes before I merge?

**Comment by swade1987** [comment]
Closing as using https://github.com/weaveworks/flux/pull/1918 now due to squash merged issue.

**Comment by squaremo** [comment]
Future reference: `git reset` to the branch commit, then `git commit`, would be a simple way to squash the commits into one. (You could do `git rebase -i <branch commit>` if you wanted to selectively squash things).

**Pull request closed without merging**